src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ApplyChangesToWorkspaceContext.cs

```diff
@@ -69,12 +73,31 @@ public async Task ApplyProjectBuildAsync(IProjectVersionedValue<IProjectSubscrip
             {
                 IComparable version = GetConfiguredProjectVersion(update);
 
-                ProcessOptions(projectChange.After);
+                ProcessOptions(buildSnapshot);
                 await ProcessCommandLineAsync(version, projectChange.Difference, state, cancellationToken);
                 ProcessProjectBuildFailure(projectChange.After);
             }
         }
 
+        private void ProcessOptions(IProjectBuildSnapshot buildSnapshot)
+        {
+            Assumes.NotNull(_context);
+
+            buildSnapshot.TargetOutputs.TryGetValue(
+                "CompileDesignTime",
+                out IImmutableList<KeyValuePair<string, IImmutableDictionary<string, string>>> targetOutputs);
+
+            var options = ImmutableArray.CreateBuilder<string>(targetOutputs.Count);
```

Comment:
What makes TargetsOuputs.Count a good capacity here?

Reply:
Yes, in order to `Move` we need to get the count right 100% of the time, otherwise it'll throw. There isn't a more efficient way than this.